### PR TITLE
upload-api carStoreBucket#has checks r2

### PR DESCRIPTION
**Pull request by gobengo**
Part of
* https://github.com/web3-storage/w3up/issues/1300

### CODE REVIEW

File: upload-api/buckets/car-store.js [hunk 1]
```diff
@@ -81,3 +81,39 @@ export function useCarStore(s3, bucketName) {
...
+
+/**
+ * compose many car stores.
+ * store#createUploadUrl is from first store.
+ * store#has will check stores in order until 0-1 `true` are found.
+ * 
+ * @param  {import('@web3-storage/upload-api').CarStoreBucket} carStore 
+ * @param  {Array<import('@web3-storage/upload-api').CarStoreBucket>} moreCarStores 
+ */
+export function composeCarStores(carStore, ...moreCarStores) {
```
Comment (vasco-santos) [inline upload-api/buckets/car-store.js:93]: I would call it something else, where is more clear to it fallbacking only on `has`, being everything else the first. More a `fallbackHasComposeCarStore`? not sure :)

File: upload-api/functions/ucan-invocation-router.js [hunk 2]
```diff
@@ -141,7 +145,16 @@ export async function ucanInvocationRouter(request) {
     storeTable: createStoreTable(AWS_REGION, storeTableName, {
       endpoint: dbEndpoint,
     }),
-    carStoreBucket: createCarStore(AWS_REGION, storeBucketName),
+    carStoreBucket: composeCarStores(
+      createCarStore(AWS_REGION, storeBucketName),
+      createCarStore('auto', carparkBucketName, {
```
Comment (vasco-santos) [inline upload-api/functions/ucan-invocation-router.js:150]:
```suggestion
      createCarStore(R2_REGION, carparkBucketName, {
```

File: upload-api/functions/ucan-invocation-router.js [hunk 3]
```diff
@@ -262,5 +275,9 @@ function getLambdaEnv () {
     accessServiceURL: mustGetEnv('ACCESS_SERVICE_URL'),
     // set for testing
     dbEndpoint: process.env.DYNAMO_DB_ENDPOINT,
```
Comment (vasco-santos) [inline upload-api/functions/ucan-invocation-router.js:277]: nit: leave this into the end as it is a testing only thing :)

**Review by vasco-santos — COMMENTED** [review]
strategy here looks great to me :) Would just suggest getting the tests to run with testcontainers

File: upload-api/buckets/car-store.js (hunk 1 above)
Comment (gobengo) [reply upload-api/buckets/car-store.js:93]: renaming to `composeCarStoresWithOrderedHas`

File: upload-api/buckets/car-store.js [hunk 4]
```diff
@@ -103,3 +103,55 @@ export function useCarStore(s3, bucketName) {
...
+
+/**
+ * compose many car stores.
+ * store#createUploadUrl is from first store.
+ * store#has will check stores in order until 0-1 `true` are found.
+ * 
+ * @param  {import('@web3-storage/upload-api').CarStoreBucket} carStore 
+ * @param  {Array<import('@web3-storage/upload-api').CarStoreBucket>} moreCarStores 
+ */
+export function composeCarStoresWithOrderedHas(carStore, ...moreCarStores) {
```
Comment (gobengo) [inline upload-api/buckets/car-store.js:115]: if this name still feels wrong after @vasco-santos pointed it out, honestly happy to just accept the first suggestion

File: upload-api/test/ucan-invocation.test.js [hunk 5]
```diff
@@ -443,7 +443,8 @@ test('can process ucan log request for given receipt after its invocation stored
   const message = await CAR.request.decode(workflow)
 
   const invocationCid = message.invocations[0].cid
-  t.deepEqual(invocationCid, invocation.link())
+
+  t.deepEqual(invocationCid.bytes, invocation.link().bytes)
```
Comment (gobengo) [inline upload-api/test/ucan-invocation.test.js:447]: this was to work around some very slight mismatch with a new property `[Symbol.toStringTag]` for CID. still trying to figure out where that comes from, but I figure this is still a pretty good deepEqual for a cid right?

**Comment by gobengo** [comment]
> Would just suggest getting the tests to run with testcontainers

@vasco-santos I added some commits that I think does what you mean by this? lmk if there's more that makes sense

**Review by vasco-santos — APPROVED** [review]
LGTM, small nits on comments

File: upload-api/test/helpers/ucan.js [hunk 6]
```diff
@@ -195,7 +196,12 @@ export async function executionContextToUcantoTestServerContext (t) {
     dynamo,
     await createTable(dynamo, uploadTableProps)
   );
-  const carStoreBucket = useCarStore(s3, bucketName);
+  const carParkBucket = carParkBucketName ? useCarStore(r2, carParkBucketName) : undefined
```
[the remark this answers is not in the document]
Comment (gobengo) [reply upload-api/test/helpers/ucan.js]: I renamed it to start with `r2` good idea

**Pull request merged**